Helper/Cart.php

```diff
@@ -587,12 +587,39 @@ protected function getCartCacheIdentifier($cart)
         // extend cache identifier with custom address fields
         $immutableQuote = $this->getLastImmutableQuote();
         $identifier .= $this->convertCustomAddressFieldsToCacheIdentifier($immutableQuote);
+        $identifier .= $this->convertExternalFieldsToCacheIdentifier($immutableQuote);
 
+        return md5($identifier);
+    }
+
+    /**
+     * @param $immutableQuote
+     * @return string
+     */
+    private function convertExternalFieldsToCacheIdentifier($immutableQuote)
+    {
+        $cacheIdentifier = "";
+        // add gift message id into cart cache identifier
         if($giftMessageId = $immutableQuote->getGiftMessageId()) {
-            $identifier .= $giftMessageId;
+            $cacheIdentifier .= $giftMessageId;
         }
 
-        return md5($identifier);
+        // add gift wrapping id into cart cache identifier
+        if ($giftWrappingId = $immutableQuote->getGwId()) {
+            $cacheIdentifier .= $giftWrappingId;
+        }
+
+        // add gift wrapping item ids into cart cache identifier
+        $quoteItems = $immutableQuote->getAllVisibleItems();
```

Comment:
This call costs something. Could you check if we have any way to move the calculation below into bolt cart calculation?

Reply:
@vitaliyreznikov  I don’t think this call costs anything as the Core Magento Quote model already handles the caching logic to get the quote items and it also uses that call in a lot of other places. See here - https://prnt.sc/tt9qny https://prnt.sc/tt9w6c